### PR TITLE
Asset Browser tweaks

### PR DESCRIPTION
**Pull request by Atlante45**
- Added information box for baking
- Added tooltip of elided filenames
- Added bakes pending count
- Added multi-select enabling/disabling for baking

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3663/HighFidelity-Beta-PR11471-73483eef945e4d1a419d6b05977cda28661a3e91.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3663/HighFidelity-Beta-PR11471-73483eef945e4d1a419d6b05977cda28661a3e91.exe

File: interface/resources/qml/hifi/AssetServer.qml [hunk 1]
```diff
@@ -584,8 +596,24 @@ ScrollingWindow {
...
+                            
+                            acceptedButtons: Qt.NoButton
+                            hoverEnabled: true
+
+                            onEntered: {
+                                if (parent.truncated) {
+                                    treeLabelToolTip.show(parent);
+                                }
+                            }
+                            onExited: treeLabelToolTip.hide();
```
Comment (AndrewMeadows) [inline interface/resources/qml/hifi/AssetServer.qml]: I see some code in this PR that uses a comma `,` at the end of each map entry, or a semicolon `;`, and sometimes nothing.  I take it that semicolons (and commas) are optional in QML as they are in JS.  Do we have a convention that we try to maintain?  If so, which is it?

File: interface/src/scripting/AssetMappingsScriptingInterface.h [hunk 2]
```diff
@@ -38,10 +39,13 @@ class AssetMappingModel : public QStandardItemModel {
     bool isKnownMapping(QString path) const { return _pathToItemMap.contains(path); }
     bool isKnownFolder(QString path) const;
 
+    int getBakesPendingCount() const { return _bakesPendingCount;  }
```
Comment (AndrewMeadows) [inline interface/src/scripting/AssetMappingsScriptingInterface.h]: Maybe should be: `getNumPendingBakes()` and `_numPendingBakes`?  Those would agree more with our coding standard [section 1.2.7](https://github.com/highfidelity/hifi/wiki/Coding-Standard#127-the-prefix-num-should-be-used-for-variables-representing-a-number-of-objects).

File: interface/resources/qml/hifi/AssetServer.qml (hunk 1 above)
Comment (Atlante45) [reply interface/resources/qml/hifi/AssetServer.qml]: `,` are for lists, I don't think you can use them at the end of statements.
For `;` vs no `;`, I don't know if we have an actual standard for that.
I'll tend to match the code I'm working in.

**Comment by Atlante45** [comment]
build this please

**Comment by Atlante45** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3716/HighFidelity-Beta-PR11471-eb06c33187533d5bb8a822a32cfc621e9424640f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3716/HighFidelity-Beta-PR11471-eb06c33187533d5bb8a822a32cfc621e9424640f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3754/HighFidelity-Beta-PR11471-af69222fdb236d041db696660ba49dadd34d2463.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3754/HighFidelity-Beta-PR11471-af69222fdb236d041db696660ba49dadd34d2463.exe

**Comment by Atlante45** [comment]
## Test Plan:
- Make sure the information box works on both the desktop and tablet Asset browser.
   (Click the icon next to the "Use baked" checkbox)
- Rename an asset to something longer than the size of the window
    - Hovering over it should expand the name in a tooltip
- If you are not multi-selecting and assets are pending, the "x items selected" should change to "x items pending"
    - You can use the following feature to trigger that case
- Multi-select assets with the same baking status you should be able to click the checkbox to change all of them at once.
    - If they do not have the same status, but box should be deactivated.

**Comment by OmegaHeron** [comment]
Will test.

**Comment by OmegaHeron** [comment]
Tested Windows 10 (Desktop and CV1)

@Atlante45  you say;

`If you are not multi-selecting and assets are pending, the "x items selected" should change to "x items pending"  Multi-select assets with the same baking status you should be able to click the checkbox to change all of them at once. If they do not have the same status, but box should be deactivated.`

How do you multi-select items in ATP Browser?  I've *never* been able to select more than one item in it.  Shift + Click does not, nor Ctrl+Click nor anything else I've tried.

---

Make sure the information box works on both the desktop and tablet Asset browser.
(Click the icon next to the "Use baked" checkbox)

**Pass**

---

Rename an asset to something longer than the size of the window
Hovering over it should expand the name in a tooltip -  

**Pass**/Yes, but...

Tool tip is still no wider than maximal length of ATP overlay dialog (which is what I'd expect) so there could be cases where, as worded, this would be a fail, but, it's not.

**Comment by Atlante45** [comment]
@OmegaHeron You need to right click to multi-select.
I know that's not intuitive at all and I'd like to get that behavior fixed (Don't shoot the messenger!).
Haven't had time to look into it.

For the tooltip, agreed, getting it to show outside of the bounds of the window is tricky with our current setup though.

**Comment by OmegaHeron** [comment]
Ok @atlante45 - I did right click, but that gives a pop up menu for copy url etc etc.  Let me try again and see if I can do multi-sel with that.

**Comment by OmegaHeron** [comment]
This is a QA Pass (Windows 10) for specific function of added function via test plan.

But -- it's next to impossible to use multi-select with right click button as selector (My - turn to say, don't shoot messenger!).

**Strongly suggest this be a work in progress until refined vs merging into dev and/or having this turn up in RC as is.**

Each right click selects an item and gives a pop-up menu so it gets really really messy.

A second right click does not deselect a member or multiple selection (only way is to left click and start over without an item you may not have wanted in multi select).

It's probably not optimal that you can add a "folder" to multi select.

If stack terminates (for a restart) you go into an endless loop of messages that Interface can't get a list of ATP items (I don't think this is unique to this PR), but is new compared to my memory of ATP browser operation from days past.  I.E. It used to gracefully handle ATP server going away by simply closing ATP browser or showing an empty list vs spewing warning dialog that repeats once dismissed.

**Comment by Atlante45** [comment]
@OmegaHeron Note that multi-select has been here for a while, but I completely agree.
I've prototyped a bit of it over the weekend but could not get it over the finish line.

For the folder, I disagree. Multi-select was originally added to be able to delete multiple things at once. and you would definitely want to be able to delete a folder.
For other operations, they disable themselves if they detect that the set of selected items is not suitable for said operation.

The messages are relatively new because we added auto-updates (You don't have to manually refresh anymore).
You shouldn't get and endless loop though.
We have a ticket filed on this and I thought it was fixed. Will look into it.

**Comment by OmegaHeron** [comment]
@Atlante45 - Fair enough and makes sense in said context.  Figure out a way to use something other than right click and you've got a winner.

**Comment by Atlante45** [comment]
Yeah, I started implementing typical Ctrl/Shift support, but I couldn't get QML to behave.
